optuna/samplers/__init__.py

```diff
@@ -44,3 +47,32 @@ def product_search_space(study):
             del search_space[param_name]
 
     return search_space or {}
+
+
+def product_search_space(study):
+    # type: (BaseStudy) -> Dict[str, BaseDistribution]
+    """Return the product search space of the :class:`~optuna.study.BaseStudy`.
+
+    **[NOTICE]**
+    This function is deprecated.
+    Please use :func:`~optuna.samplers.intersection_search_space` instead.
+
+    Product search space contains the product set of parameter distributions that have been
+    suggested in the completed trials of the study so far.
+    If there are multiple parameters that have the same name but different distributions,
+    neither is included in the resulting search space
+    (i.e., the parameters with dynamic value ranges are excluded).
```

Comment:
IMO, we can just mention that this function is an alias for `intersection_search_space` instead of fully explaining the functionality because this function will be deleted in the future.